PrebidMobile/PrebidMobile-core/src/main/java/org/prebid/mobile/PrebidNativeAd.java

```diff
@@ -0,0 +1,260 @@
+                    JSONObject adObject = asset.getJSONObject(i);
+                    if (adObject.has("title")) {
+                        JSONObject title = adObject.getJSONObject("title");
+                        if (title.has("text")) {
+                            ad.setTitle(title.getString("text"));
+                        }
+                    }
+                    if (adObject.has("description")) {
+                        JSONObject description = adObject.getJSONObject("description");
+                        if (description.has("text")) {
+                            ad.setDescription(description.getString("text"));
+                        }
+                    }
+
+                    if (adObject.has("img")) {
+                        JSONObject img = adObject.getJSONObject("img");
+                        if (img.has("url")) {
+                            ad.setImageUrl(img.getString("url"));
+                        }
+                    }
+
+                    if (adObject.has("icon")) {
+                        JSONObject icon = adObject.getJSONObject("icon");
+                        if (icon.has("url")) {
+                            ad.setIconUrl(icon.getString("url"));
+                        }
+                    }
+
+                    ad.setCallToAction(details.optString("cta"));
+
+//                    ad.setClickUrl(details.getString("clickUrl"));
+                }
+
+                if (details.has("link")) {
+                    JSONObject link = details.getJSONObject("link");
+                    if (link.has("url")) {
+                        ad.setClickUrl(link.getString("url"));
+                    }
+                }
+
+                if (details.has("eventtrackers")) {
+                    JSONArray eventtrackers = details.getJSONArray("eventtrackers");
+                    if (eventtrackers.length() > 0) {
+                        ad.imp_trackers = new ArrayList<>();
+                        for (int count = 0; count < eventtrackers.length(); count++) {
+                            JSONObject eventtracker = eventtrackers.getJSONObject(count);
+                            if (eventtracker.has("url")) {
+                                ad.imp_trackers.add(eventtracker.getString("url"));
+                            }
+                        }
+                    }
+                }
+//                ad.imp_trackers =
+                ad.anNativeExpireHandler = new Handler(Looper.getMainLooper());
+                ad.anNativeExpireHandler.postDelayed(ad.expireRunnable, NATIVE_AD_RESPONSE_EXPIRATION_TIME);
+                return ad.isValid() ? ad : null;
+            } catch (JSONException e) {
+                e.printStackTrace();
+            }
+        }
+        return null;
+    }
+
+    private boolean isValid() {
+        return URLUtil.isValidUrl(clickUrl) &&
+                !TextUtils.isEmpty(title) &&
+//                !TextUtils.isEmpty(description) &&
```

Comment:
Why commenting out these?

Reply:
Those were commented for manual testing purpose.